File: lobster-platform/upload-critical.js
const https = require('https');
const fs = require('fs');
const path = require('path');

const TOKEN = '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';
const OWNER = 'D2758695161';
const REPO = 'wander-lobster-platform';
const OUT = 'out';

function getFilePaths(dir) {
  const results = [];
  const items = fs.readdirSync(dir);
  for (const item of items) {
    const full = path.join(dir, item);
    if (fs.statSync(full).isDirectory()) {
      if (item === '.next' || item === 'node_modules') continue;
      results.push(...getFilePaths(full));
    } else {
      results.push(full.slice(OUT.length + 1));
    }
  }
  return results;
}

const files = getFilePaths(OUT);
console.log('Found', files.length, 'files to upload');

// Upload first 10 critical ones
const critical = files.slice(0, 10);
console.log('Critical files:', critical);

let idx = 0;
let ok = 0;
let fail = 0;

function uploadOne(file, cb) {
  const filePath = path.join(OUT, file);
  if (!fs.existsSync(filePath)) { cb(); return; }
  const content = fs.readFileSync(filePath);
  const b64 = content.toString('base64');

  // Get SHA first
  const getOpts = {
    hostname: 'api.github.com',
    path: '/repos/' + OWNER + '/' + REPO + '/contents/' + file,
    method: 'GET',
    headers: { 'Authorization': 'Bearer ' + TOKEN, 'User-Agent': 'yitong', 'Accept': 'application/vnd.github.v3+json' }
  };
  let sha = null;
  const getReq = https.get(getOpts, (res) => {
    let d = '';
    res.on('data', c => d += c);
    res.on('end', () => {
      try {
        const j = JSON.parse(d);
        sha = j.sha;
      } catch(e) {}
      doPut(sha);
    });
  });
  getReq.on('error', () => doPut(null));

  function doPut(sha) {
    const body = JSON.stringify({
      message: 'upload ' + file,
      content: b64,
      sha: sha || undefined
    });
    const putOpts = {
      hostname: 'api.github.com',
      path: '/repos/' + OWNER + '/' + REPO + '/contents/' + file,
      method: 'PUT',
      headers: { 'Authorization': 'Bearer ' + TOKEN, 'User-Agent': 'yitong', 'Content-Type': 'application/json', 'Content-Length': Buffer.byteLength(body) }
    };
    const req = https.request(putOpts, (res) => {
      let d = '';
      res.on('data', c => d += c);
      res.on('end', () => {
        if (res.statusCode >= 200 && res.statusCode < 300) {
          ok++;
          console.log('OK', ok + '/' + files.length, file);
        } else {
          fail++;
          console.log('FAIL', file, res.statusCode, d.slice(0, 100));
        }
        cb();
      });
    });
    req.on('error', (e) => {
      fail++;
      console.log('ERR', file, e.message);
      cb();
    });
    req.write(body);
    req.end();
  }
}

function runNext() {
  if (idx >= critical.length) {
    console.log('\nCritical upload done. OK:', ok, 'Fail:', fail);
    console.log('Now uploading remaining', files.length - critical.length, 'files in background...');
    // Continue with rest in background
    let i2 = critical.length;
    function uploadRest() {
      if (i2 >= files.length) {
        console.log('\nAll done. Total OK:', ok, 'Fail:', fail);
        return;
      }
      uploadOne(files[i2++], uploadRest);
    }
    uploadRest();
    return;
  }
  uploadOne(critical[idx++], runNext);
}

runNext();

File: lobster-platform/upload-critical.py
import os
import json
import base64
import http.client
import time

TOKEN = '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'
OWNER = 'D2758695161'
REPO = 'wander-lobster-platform'
OUT = 'out'

def get_file_paths(directory):
    results = []
    for root, dirs, files in os.walk(directory):
        dirs[:] = [d for d in dirs if d not in ('.next', 'node_modules')]
        for f in files:
            full = os.path.join(root, f)
            rel = full[len(OUT)+1:]
            results.append(rel)
    return results

def upload_file(file_path):
    full = os.path.join(OUT, file_path)
    if not os.path.exists(full):
        return True, None
    
    with open(full, 'rb') as f:
        content = f.read()
    b64 = base64.b64encode(content).decode('ascii')
    
    # Get SHA
    conn = http.client.HTTPSConnection('api.github.com', timeout=10)
    headers = {
        'Authorization': f'Bearer {TOKEN}',
        'User-Agent': 'yitong',
        'Accept': 'application/vnd.github.v3+json'
    }
    sha = None
    try:
        conn.request('GET', f'/repos/{OWNER}/{REPO}/contents/{file_path}', headers=headers)
        r = conn.getresponse()
        if r.status == 200:
            data = json.loads(r.read())
            sha = data.get('sha')
    except:
        pass
    finally:
        conn.close()
    
    # Upload
    body = json.dumps({
        'message': f'upload {file_path}',
        'content': b64,
        'sha': sha
    })
    
    conn2 = http.client.HTTPSConnection('api.github.com', timeout=15)
    try:
        conn2.request('PUT', f'/repos/{OWNER}/{REPO}/contents/{file_path}', body=body, headers={
            **headers,
            'Content-Type': 'application/json'
        })
        r = conn2.getresponse()
        data = r.read()
        if 200 <= r.status < 300:
            return True, None
        else:
            return False, f'{r.status} {data[:100].decode()}'
    except Exception as e:
        return False, str(e)
    finally:
        conn2.close()

files = get_file_paths(OUT)
print(f'Found {len(files)} files')

# Critical first
critical = [f for f in files if any(x in f for x in ['index.html', 'shop.html', 'aiwallet', 'kyc'])]
print(f'Critical: {critical[:10]}')

ok = 0
fail = 0
for i, f in enumerate(critical[:15]):
    ok2, err = upload_file(f)
    if ok2:
        ok += 1
        print(f'OK {ok}/{len(files)} {f}')
    else:
        fail += 1
        print(f'FAIL {f} {err}')
    time.sleep(0.3)

print(f'\nCritical done: OK={ok} Fail={fail}')
print('Uploading rest in background...')

# Upload rest in batches of 5
rest = [f for f in files if f not in critical[:15]]
batch_size = 5
for batch_start in range(0, len(rest), batch_size):
    batch = rest[batch_start:batch_start+batch_size]
    for f in batch:
        ok2, err = upload_file(f)
        if ok2:
            ok += 1
        else:
            fail += 1
        print(f'OK={ok} FAIL={fail} {f}')
        time.sleep(0.2)
    time.sleep(1)

print(f'\nAll done: OK={ok} Fail={fail}')
